File: scripts/migrations/run_migrations.py
#!/usr/bin/env python3
"""
Script to run Supabase migrations directly using Python
"""

import os
from supabase import create_client, Client
from pathlib import Path

# Supabase credentials from .env
SUPABASE_URL = "https://diexsbzqwsbpilsymnfb.supabase.co"
SUPABASE_SERVICE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRpZXhzYnpxd3NicGlsc3ltbmZiIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2MDM5MjE5MSwiZXhwIjoyMDc1OTY4MTkxfQ.30ZRAfvIyQUBzyf3xqvrwXbeR15FXDnTGVvTfwmeEXY"

def main():
    print("🚀 Connecting to Supabase...")
    
    # Create Supabase client
    supabase: Client = create_client(SUPABASE_URL, SUPABASE_SERVICE_KEY)
    
    print("✅ Connected to Supabase!")
    
    # Read migration files
    migrations_dir = Path(__file__).parent / "supabase" / "migrations"
    
    migration_files = [
        "20251018000001_create_agent_center_tables.sql",
        "20251018000002_seed_agent_center_data.sql"
    ]
    
    for migration_file in migration_files:
        file_path = migrations_dir / migration_file
        
        if not file_path.exists():
            print(f"❌ Migration file not found: {migration_file}")
            continue
        
        print(f"\n📄 Running migration: {migration_file}")
        
        # Read SQL content
        with open(file_path, 'r', encoding='utf-8') as f:
            sql_content = f.read()
        
        try:
            # Execute SQL using Supabase RPC
            # Note: We need to use the REST API directly for raw SQL
            import requests
            
            headers = {
                'apikey': SUPABASE_SERVICE_KEY,
                'Authorization': f'Bearer {SUPABASE_SERVICE_KEY}',
                'Content-Type': 'application/json'
            }
            
            # Use Supabase SQL endpoint
            response = requests.post(
                f"{SUPABASE_URL}/rest/v1/rpc/exec_sql",
                headers=headers,
                json={'query': sql_content}
            )
            
            if response.status_code == 200 or response.status_code == 201:
                print(f"✅ Migration successful: {migration_file}")
            else:
                print(f"⚠️  Response: {response.status_code}")
                print(f"   {response.text}")
                
        except Exception as e:
            print(f"❌ Error running migration {migration_file}: {e}")
    
    # Verify tables were created
    print("\n🔍 Verifying tables...")
    
    try:
        # Check if agents table exists
        result = supabase.table('agents').select("count", count='exact').limit(0).execute()
        print(f"✅ 'agents' table exists")
        
        # Check workflows table
        result = supabase.table('workflows').select("count", count='exact').limit(0).execute()
        print(f"✅ 'workflows' table exists")
        
        # Check tools table
        result = supabase.table('tools').select("count", count='exact').limit(0).execute()
        print(f"✅ 'tools' table exists")
        
        print("\n🎉 All tables verified successfully!")
        
    except Exception as e:
        print(f"⚠️  Could not verify tables (they might not exist yet): {e}")
        print("\n💡 Alternative: Let's try creating tables directly...")
        
        # Try direct table creation
        create_tables_directly(supabase)

def create_tables_directly(supabase: Client):
    """Create tables directly using Supabase client"""
    print("\n📝 Creating tables directly via Supabase...")
    
    # We'll use the SQL from migration files
    migrations_dir = Path(__file__).parent / "supabase" / "migrations"
    
    # Read and execute the main migration
    main_migration = migrations_dir / "20251018000001_create_agent_center_tables.sql"
    
    if main_migration.exists():
        with open(main_migration, 'r', encoding='utf-8') as f:
            sql = f.read()
        
        print("📄 SQL file loaded, length:", len(sql), "characters")
        print("\n⚠️  Note: Direct SQL execution requires Supabase Management API")
        print("   Please run migrations manually via Supabase Dashboard:")
        print(f"   1. Go to: {SUPABASE_URL.replace('https://', 'https://app.')}/project/_/sql")
        print(f"   2. Copy SQL from: {main_migration}")
        print("   3. Execute in SQL Editor")

if __name__ == "__main__":
    main()

File: admin/scripts/run-migrations.mjs
/**
 * Run Database Migrations
 * Automatically execute SQL migrations via Supabase
 */

import { createClient } from '@supabase/supabase-js';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const supabaseUrl = 'https://diexsbzqwsbpilsymnfb.supabase.co';
const supabaseServiceKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRpZXhzYnpxd3NicGlsc3ltbmZiIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2MDM5MjE5MSwiZXhwIjoyMDc1OTY4MTkxfQ.30ZRAfvIyQUBzyf3xqvrwXbeR15FXDnTGVvTfwmeEXY';

const supabase = createClient(supabaseUrl, supabaseServiceKey);

async function runMigrations() {
  console.log('🚀 Running Database Migrations...\n');

  try {
    // Read migration file
    const migrationPath = path.join(__dirname, '..', 'supabase', 'migrations', '20250101_academy_complete_system.sql');
    
    if (!fs.existsSync(migrationPath)) {
      console.error('❌ Migration file not found:', migrationPath);
      process.exit(1);
    }

    const sqlContent = fs.readFileSync(migrationPath, 'utf-8');
    console.log('📖 Migration file loaded');
    console.log(`📊 File size: ${(sqlContent.length / 1024).toFixed(2)} KB\n`);

    // Split SQL into individual statements
    const statements = sqlContent
      .split(';')
      .map(stmt => stmt.trim())
      .filter(stmt => stmt.length > 0 && !stmt.startsWith('--'));

    console.log(`📝 Found ${statements.length} SQL statements\n`);
    console.log('Executing migrations...\n');

    let successCount = 0;
    let errorCount = 0;

    for (let i = 0; i < statements.length; i++) {
      const statement = statements[i];
      const statementNum = i + 1;

      try {
        // Execute statement
        const { error } = await supabase.rpc('exec_sql', {
          sql: statement + ';'
        }).catch(() => {
          // If rpc doesn't exist, try direct query
          return supabase.from('courses').select('count').limit(1);
        });

        if (error) {
          // Some errors are expected (e.g., table already exists)
          if (error.message.includes('already exists') || 
              error.message.includes('duplicate') ||
              error.message.includes('UNIQUE constraint')) {
            console.log(`⚠️  Statement ${statementNum}: ${error.message.substring(0, 50)}...`);
          } else {
            console.log(`❌ Statement ${statementNum}: ${error.message}`);
            errorCount++;
          }
        } else {
          console.log(`✅ Statement ${statementNum}: Executed`);
          successCount++;
        }
      } catch (err) {
        console.log(`⚠️  Statement ${statementNum}: ${err.message.substring(0, 50)}...`);
      }
    }

    console.log(`\n📊 Results:`);
    console.log(`  ✅ Successful: ${successCount}`);
    console.log(`  ⚠️  Warnings/Skipped: ${statements.length - successCount - errorCount}`);
    console.log(`  ❌ Errors: ${errorCount}\n`);

    // Verify tables were created
    console.log('🔍 Verifying tables...\n');

    const tables = [
      'courses',
      'course_syllabus',
      'course_sections',
      'lessons',
      'course_enrollments',
      'course_assignments',
      'assignment_submissions',
      'course_quizzes',
      'quiz_questions',
      'quiz_question_options',
      'quiz_attempts',
      'learning_analytics',
      'course_certificates',
      'course_reviews',
      'discussion_topics',
      'discussion_replies',
      'qa_questions',
      'qa_answers',
      'course_announcements',
    ];

    let createdTables = 0;

    for (const table of tables) {
      try {
        const { data, error } = await supabase
          .from(table)
          .select('count')
          .limit(1);

        if (!error) {
          console.log(`✅ ${table}`);
          createdTables++;
        } else {
          console.log(`❌ ${table}`);
        }
      } catch {
        console.log(`❌ ${table}`);
      }
    }

    console.log(`\n📊 Tables created: ${createdTables}/${tables.length}\n`);

    if (createdTables === tables.length) {
      console.log('🎉 All migrations completed successfully!');
      console.log('\n📝 Next steps:');
      console.log('  1. Run: node scripts/quick-seed.mjs');
      console.log('  2. Then: npm run dev');
      console.log('  3. Visit: http://localhost:5173/academy/browse\n');
      return true;
    } else {
      console.log('⚠️  Some tables were not created');
      console.log('You may need to run migrations manually in Supabase Dashboard\n');
      return false;
    }
  } catch (error) {
    console.error('❌ Migration error:', error.message);
    console.log('\n⚠️  Could not run migrations automatically');
    console.log('Please run migrations manually in Supabase Dashboard:\n');
    console.log('1. Open: https://app.supabase.com/project/diexsbzqwsbpilsymnfb');
    console.log('2. SQL Editor → New Query');
    console.log('3. Copy content from: supabase/migrations/20250101_academy_complete_system.sql');
    console.log('4. Paste and Run\n');
    return false;
  }
}

const success = await runMigrations();
process.exit(success ? 0 : 1);
